crates/ruff_python_literal/src/format.rs

```diff
@@ -282,17 +314,68 @@ fn parse_precision(text: &str) -> Result<(Option<usize>, &str), FormatSpecError>
     })
 }
 
+/// Parses a format part within a format spec
+fn parse_nested_format_part<'a>(
+    parts: &'a RefCell<Vec<FormatPart>>,
```

Comment:
You can avoid `RefCell` like:

```rust
diff --git a/crates/ruff_python_literal/src/format.rs b/crates/ruff_python_literal/src/format.rs
index ed56457ef..6df394895 100644
--- a/crates/ruff_python_literal/src/format.rs
+++ b/crates/ruff_python_literal/src/format.rs
@@ -316,10 +316,9 @@ fn parse_precision(text: &str) -> Result<(Option<usize>, &str), FormatSpecError>
 
 /// Parses a format part within a format spec
 fn parse_nested_format_part<'a>(
-    parts: &'a RefCell<Vec<FormatPart>>,
+    parts: &mut Vec<FormatPart>,
     text: &'a str,
 ) -> Result<&'a str, FormatSpecError> {
-    let mut parts = parts.borrow_mut();
     let mut end_bracket_pos = None;
     let mut left = String::new();
 
@@ -356,25 +355,25 @@ fn parse_nested_format_part<'a>(
 
 impl FormatSpec {
     pub fn parse(text: &str) -> Result<Self, FormatSpecError> {
-        let replacements = RefCell::new(vec![]);
+        let mut replacements = vec![];
         // get_integer in CPython
-        let text = parse_nested_format_part(&replacements, text)?;
+        let text = parse_nested_format_part(&mut replacements, text)?;
         let (conversion, text) = FormatConversion::parse(text);
-        let text = parse_nested_format_part(&replacements, text)?;
+        let text = parse_nested_format_part(&mut replacements, text)?;
         let (mut fill, mut align, text) = parse_fill_and_align(text);
-        let text = parse_nested_format_part(&replacements, text)?;
+        let text = parse_nested_format_part(&mut replacements, text)?;
         let (sign, text) = FormatSign::parse(text);
-        let text = parse_nested_format_part(&replacements, text)?;
+        let text = parse_nested_format_part(&mut replacements, text)?;
         let (alternate_form, text) = parse_alternate_form(text);
-        let text = parse_nested_format_part(&replacements, text)?;
+        let text = parse_nested_format_part(&mut replacements, text)?;
         let (zero, text) = parse_zero(text);
-        let text = parse_nested_format_part(&replacements, text)?;
+        let text = parse_nested_format_part(&mut replacements, text)?;
         let (width, text) = parse_number(text)?;
-        let text = parse_nested_format_part(&replacements, text)?;
+        let text = parse_nested_format_part(&mut replacements, text)?;
         let (grouping_option, text) = FormatGrouping::parse(text);
-        let text = parse_nested_format_part(&replacements, text)?;
+        let text = parse_nested_format_part(&mut replacements, text)?;
         let (precision, text) = parse_precision(text)?;
-        let text = parse_nested_format_part(&replacements, text)?;
+        let text = parse_nested_format_part(&mut replacements, text)?;
 
         let (format_type, _text) = if text.is_empty() {
             (None, text)
@@ -406,7 +405,7 @@ impl FormatSpec {
             grouping_option,
             precision,
             format_type,
-            replacements: replacements.take(),
+            replacements,
         })
     }
 ```

Reply:
I believe the key is to avoid marking the vector as having the same lifetime as the return type. I.e., you want to _avoid_ `parts: &'a mut Vec<FormatPart>`.

Reply:
(We don't actually _need_ multiple and dynamic mutability rules here so it's better to avoid `RefCell`.)